src/py-opentimelineio/opentimelineio/url_utils.py

```diff
@@ -62,10 +65,18 @@ def filepath_from_url(urlstr):
     if PureWindowsPath(parsed_result.netloc).drive:
         filepath = Path(parsed_result.netloc + parsed_result.path)
 
-    # Otherwise check if the specified index is a windows drive, then offset the path
+    # Check if the specified index has a specified `drive`, if so then do nothing
+    elif filepath.drive:
+        filepath = filepath
+
+    # Check if the specified index is a windows drive, then offset the path
     elif PureWindowsPath(filepath.parts[1]).drive:
         # Remove leading "/" if/when `request.url2pathname` yields "/S:/path/file.ext"
         filepath = filepath.relative_to(filepath.root)
 
+    # Last resort, strip the "file:" prefix
+    else:
+        filepath = Path(decoded_url_str.strip('file:'))
+
```

Comment:
In other environments than windows or URLs without windows drives/shares , this will manipulate all URLs.

Reply:
This seems to work with the example paths for the Posix paths. Did i use invalid paths?

Reply:
Also, from what I can tell, when AVID references a UNC path on Windows it only has `//` and `///`, that's why I just stripped `file:`

Reply:
>This seems to work with the example paths for the Posix paths. Did i use invalid paths?

With the exception of possibly leaving in `localhost` in the path (more on this below), I don't think it produced invalid paths. My main point was that in a non-Windows environment this will run on all paths which seems unnecessary.
 
>Also, from what I can tell, when AVID references a UNC path on Windows it only has `//` and `///`, that's why I just stripped `file`:

Would you be able to provide example paths we can look at that show the two variations so we can dissect the paths?

From what I gather in [this](https://en.wikipedia.org/wiki/File_URI_scheme) wikipedia page there should "always" be three slashes. 
`file:///path/to/file.ext` is more or less the same as `file://localhost/path/to/file.ext` once resolved. Difference being that `localhost` is implied in the first one and defined in the second one. 
If we replace `localhost` with `SERVERNAME` we're getting closer to the UNC paths provided in your tests.

In both your and my approach will produce a path with `localhost` in it given the URL: `file://localhost/path/to/file.ext` -> `//localhost/path/to/file.ext` which I think would produce an error down the line.
Since `localhost` is "reserved" it shouldn't make it in a file path.

So whether we go for stripping the scheme or joining parts I think we should adjust the logic to:
`elif parsed_result.netloc and parsed_result.netloc != 'localhost':` (updated in comment above)
which should produce: `//WINDOWSSHARE/path/to/file.ext` but not `//localhost/path/to/file.ext`

We should perhaps add another path to test
`POSIX_LOCALHOST_URL = "file://localhost/path/sub dir/file.ext"`

Reply:
> Would you be able to provide example paths we can look at that show the two variations so we can dissect the paths?

@apetrynet, I debugged the parsing an `.aaf` and came across this section of code in `advanced_authoring_format.py` 
![image](https://github.com/AcademySoftwareFoundation/OpenTimelineIO/assets/22691827/8895d62d-6151-438f-8eb1-2930e1501de8)

Seems the `.aaf` I happened to be testing had a clip with a url like this: `file://unc/path/to/file.ext`, with only 2 `/` and not 3. I couldn't tell you why that is the case but that's what led me to make this PR to fix that.


> So whether we go for stripping the scheme or joining parts I think we should adjust the logic to:
elif parsed_result.netloc and parsed_result.netloc != 'localhost': (updated in comment above)
which should produce: //WINDOWSSHARE/path/to/file.ext but not //localhost/path/to/file.ext

I like your suggestion way better than mine and switched to it! Unfortunately, I'm getting failed unit tests with `tests.test_url_conversions.TestConversions.test_windows_unc_urls`. I'm not sure how we resolve this.

![image](https://github.com/AcademySoftwareFoundation/OpenTimelineIO/assets/22691827/76ca2262-ea0c-4f6a-b04e-6669ec32a675)

> We should perhaps add another path to test POSIX_LOCALHOST_URL = "file://localhost/path/sub dir/file.ext"

Added!

Reply:
@apetrynet, I think I've reached a solution that gets all the unit tests to pass. Let me know what you think!

Reply:
Sorry I disappeared, must of zoned off while attempting to telepathically  connect with @meshula who also had some comments earlier. I think we should go ahead as the tests pass and we cover the old behavior.